sbin/md5/md5.c

```diff
@@ -610,20 +622,11 @@ main(int argc, char *argv[])
 					rec = rec->next;
 				continue;
 			}
-			/*
-			 * XXX Enter capability mode on the last argv file.
-			 * When a casper file service or other approach is
-			 * available, switch to that and enter capability mode
-			 * earlier.
-			 */
-			if (*(argv + 1) == NULL) {
 #ifdef HAVE_CAPSICUM
-				cap_rights_init(&rights, CAP_READ, CAP_FSTAT);
-				if (caph_rights_limit(fileno(f), &rights) < 0 ||
-				    caph_enter() < 0)
-					err(1, "capsicum");
+			if (caph_rights_limit(fileno(f), &rights) < 0 ||
+			    caph_enter() < 0)
```

Comment:
I don't think we should be calling caph_enter() at all here. Now you call caph_enter_casper() before processing input, that's all that's needed.

Reply:
Makes sense. Hopefully fixed now.